ui/dashboards/src/components/Variables/VariableList.tsx

```diff
@@ -24,51 +24,197 @@ import {
   Table,
   Paper,
   TableHead,
+  Switch,
+  Typography,
+  IconButton,
 } from '@mui/material';
-import { useTemplateVariableDefinitions, useEditMode } from '../../context';
+import { VariableDefinition } from '@perses-dev/core';
+import { useImmer } from 'use-immer';
+import EyeIcon from 'mdi-material-ui/Eye';
+import PencilIcon from 'mdi-material-ui/Pencil';
+import TrashIcon from 'mdi-material-ui/TrashCan';
+
+import { useTemplateVariableDefinitions, useEditMode, useTemplateVariableActions } from '../../context';
 import { TemplateVariable } from './Variable';
+import { VariableEditForm } from './VariableEditorForm';
+
+function VariableEditor(props: {
+  variableDefinitions: VariableDefinition[];
+  onChange: (variableDefinitions: VariableDefinition[]) => void;
+  onCancel: () => void;
+}) {
+  const [variableDefinitions, setVariableDefinitions] = useImmer(props.variableDefinitions);
+  const [variableEditIdx, setVariableEditIdx] = useState<number | null>(null);
+  const isEditingVariable = variableEditIdx !== null;
+
+  const removeVariable = (index: number) => {
+    setVariableDefinitions((draft) => {
+      draft.splice(index, 1);
+    });
+  };
+
+  const addVariable = () => {
+    setVariableDefinitions((draft) => {
+      draft.push({
+        kind: 'TextVariable',
+        spec: {
+          name: 'New Variable',
+          value: '',
+        },
+      });
+    });
+  };
+
+  const toggleVariableVisibility = (index: number, visible: boolean) => {
+    setVariableDefinitions((draft) => {
+      const v = draft[index];
+      if (!v) {
+        return;
+      }
+      if (!v.spec.display) {
+        v.spec.display = {
+          hidden: false,
```

Comment:
We should probably try to make these consistent since there are a few places where things can be enabled/disabled, @cndonovan what did you propose calling this property for queries again?

Reply:
Yeah not a bad idea. Although will probably need to do this in a follow up PR since its changing the spec schema.

Reply:
for queries, the proposition from @cndonovan was to call it `is_enabled` because hidding a result of query make it unnecessary to run it.
That's not the same behaviour for the variable. You can hide it, but you still want to run it because you are using it in other variable that are displayed.

Reply:
Also, as with @AntoineThebaud we are currently implementing the support of the variable in the backend, it would be nice to have the current format of the variable with the different possible field, just to be sync between the FE and the BE.
That would reduce the amount of issues we will have when connecting the FE to the BE to store and retrieve the dashboard.